### PR TITLE
 Support default as params for function and procedure

### PR DESCRIPTION
**Pull request by forestkeeper**
Previously when we call functions/procedures, we don't support default keyword usage like :

select foofunc(default);
or
exec fooproc default

This commit support to use default keyword as function or procedure param when the default value is previously defined in create proc/func

Engine pr : https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/242

Task: BABEL-335

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp [hunk 1]
```diff
@@ -1403,16 +1403,16 @@ antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitAggregate_windowed_functio
 
 antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitDefault_expr(TSqlParser::Default_exprContext *ctx)
 {
-	TSqlParser::Expression_listContext *pctx = dynamic_cast<TSqlParser::Expression_listContext *>(ctx->parent);
-	if (!pctx || dynamic_cast<TSqlParser::Table_value_constructorContext *>(pctx->parent) == nullptr) /* if DEFAULT expression is used for VALUES ..., accept it */
-		handle(INSTR_UNSUPPORTED_TSQL_EXPRESSION_DEFAULT, ctx->DEFAULT());
+	// TSqlParser::Expression_listContext *pctx = dynamic_cast<TSqlParser::Expression_listContext *>(ctx->parent);
+	// if (!pctx || dynamic_cast<TSqlParser::Table_value_constructorContext *>(pctx->parent) == nullptr) /* if DEFAULT expression is used for VALUES ..., accept it */
+	// 	handle(INSTR_UNSUPPORTED_TSQL_EXPRESSION_DEFAULT, ctx->DEFAULT());
```
Comment (lejaokri) [inline contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp]: remove it

File: contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp [hunk 2]
```diff
@@ -1403,16 +1403,16 @@ antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitAggregate_windowed_functio
...
 	return visitChildren(ctx);
 }
 
 antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitExecute_parameter(TSqlParser::Execute_parameterContext *ctx)
 {
-	if (ctx->DEFAULT())
-		handle(INSTR_UNSUPPORTED_TSQL_EXECUTE_PARAMETER_DEFAULT, ctx->DEFAULT());
+	// if (ctx->DEFAULT())
+	// 	handle(INSTR_UNSUPPORTED_TSQL_EXECUTE_PARAMETER_DEFAULT, ctx->DEFAULT());
```
Comment (lejaokri) [inline contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp]: this one too

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 3]
```diff
@@ -3528,6 +3536,105 @@ PlTsqlMatchNamedCall(HeapTuple proctup, int nargs, List *argnames,
...
+static List*
+replace_pltsql_function_defaults(HeapTuple func_tuple, List *defaults, List *fargs)
+
+{
+	HeapTuple	bbffunctuple;
+
+	if (sql_dialect != SQL_DIALECT_TSQL)
+		return fargs;
+
+	bbffunctuple = get_bbf_function_tuple_from_proctuple(func_tuple);
```
Comment (lejaokri) [inline contrib/babelfishpg_tsql/src/hooks.c:3572]: `get_bbf_function_tuple_from_proctuple` returns `NULL` if `babelfish_dump_restore` is true.
ie: In the case when `babelfish_dump_restore` is true then this function will error out.

**Review by lejaokri — COMMENTED** [review]
I am hitting the following when I try to compile manually

```
/usr/bin/install: will not overwrite just-created ‘/local/home/klejao/pg15/postgres/share/extension/babelfishpg_common--3.2.0--3.3.0.sql’ with ‘sql/babelfishpg_common--3.2.0--3.3.0.sql’
make: *** [install] Error 1
```

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 3 above)
Comment (forestkeeper) [reply contrib/babelfishpg_tsql/src/hooks.c:3572]: yeah that make sense

**Review by lejaokri — APPROVED** [review]

**Pull request merged**